hyperspy/drawing/signal1d.py

```diff
@@ -87,6 +87,11 @@ def add_line(self, line, ax='left'):
             line.sf_lines = self.right_ax_lines
             if line.axes_manager is None:
                 line.axes_manager = self.right_axes_manager
+        if update_on_navigation:
```

Comment:
I think that this may not be needed, see below.

Reply:
Done.